src/models/subscription.ts

```diff
@@ -194,6 +194,15 @@ export default class Subscription extends Sequelize.Model {
 	// This is a workaround to fix a long standing bug in sequelize for JSON data types
 	// https://github.com/sequelize/sequelize/issues/4387
 	async updateSyncState(updatedState: RepoSyncStateObject): Promise<Subscription> {
+
+		// Zero repos, means zero syncs required so set COMPLETE.
+		if(!Object.keys(updatedState?.repos || {}).length) {
+			await this.update({
+				syncStatus: SyncStatus.COMPLETE
+			});
+			return this;
```

Comment:
You definitely don't want to `return this` as you'll need the `RepoSyncstate` update below to do it's job.  Just removing this will then work as it should.